Makefile

```diff
@@ -318,10 +342,15 @@ docker-scanner-plugins: ## Build scanner plugin container images
 ##@ Code generation
 
 .PHONY: gen
-gen: gen-api gen-bicep gen-helm-docs ## Generating all code, manifests, docs
+gen: gen-api-js gen-api-go gen-bicep gen-helm-docs ## Generating all code, manifests, docs
+
+.PHONY: gen-api-js
+gen-api-js: ## Generating Javascript library for API specification
+	$(info Generating API for UI code ...)
+	@(cd ui && npx @openapitools/openapi-generator-cli generate -i ../api/openapi.yaml -g typescript-axios -o ./src/api/generated --openapi-normalizer SET_TAGS_FOR_ALL_OPERATIONS=VMClarity)
```

Comment:
This should be added to package.json as an npm command to make it easy for UI developers to (re)generate the api and hide the complexity of generating the api. It doesn't belong to the make file in my opinion.
Here you should only run the appropriate npm command e.g. `cd ui && npm run generate-api`
Probably it should be added as a postinstall script to the npm package.json as well because the UI can't be built if you don't run it before the build. e.g. `"postinstall": npm run generate-api"`